### DISCUSSION TITLE
[New] Utiliser les outils de WebPerf

### DISCUSSION THREAD

**Discussion by ddebarbieux** [cnumr/best-practices#109] [Bonnes pratiques] [answered]
Systématiquement utiliser les outils de WebPerf dans les dévelopements, former les équipes à l'utilisation de ses outils.

En pratique, optimiser la WebPerf conduit souvent à améliorer l'éco-conception.

Exemple d'outils : lighthouse

**Comment by tbroyer** [comment]
…sauf quand ces outils recommandent de faire du preconnect/prefetch/preload/prerender (faire un preload d'une resource qui sera de toutes façons utilisée dans la page elle-même, c'est de l'optimisation de performance, ce n'est ni bon ni mauvais d'un point de vu de l'écoconception ; faire un preload d'une autre page, ou d'une resource qui sera utilisée sur "les autres pages", ou qui *pourrait* être utilisée sur la page elle-même en fonction des actions de l'utilisateur, c'est plutôt une mauvaise pratique d'écoconception)
Mais globalement oui il y a beaucoup de recoupements entre les bonnes pratiques d'écoconception (d'un point de vue technique) et les bonnes pratiques de performance.

**Reply by tbroyer** [reply]
J'ai ajouté https://github.com/cnumr/best-practices/discussions/158 pour ces pré-chargements.

**Comment by ACTLEM** [comment]
Cela rejoint la discussion https://github.com/cnumr/best-practices/discussions/36 dans l'état d'esprit. On est plus sur de l'outillage que sur une recommandation d'écoconception.

**Reply by devcrafting** [reply]
Ca pourrait rentrer dans le cadre de la qualité logicielle discutée dans #107

**Comment by davidgroult** [comment]
En effet il y a des points interessants à prendre en compte comme dans lighthouse avec la partie accessibilité ou SEO
Mais comme le souligne Thomas, il y a toujours du bon et du moins bon !

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/138

**Comment by EmmanuelDemey** [comment]
Pouvons nous peut être prédéfinir un preset pour la CLI de lighthouse, afin de désactiver certaines règles ?

**Reply by tbroyer** [reply]
Oui, et sauf erreur il devrait aussi être possible d'ajouter des règles à Lighthouse pour celles qui manqueraient.
